book/src/getting-started/6-wasm-builds.md

```diff
@@ -0,0 +1,154 @@
+
+The `wasm_multithread` toolchain available in 6.5 and later is currently not supported. For more information, see the [Known Issues](#known-issues) section at the bottom of this page.
+
+Info about other Qt and emscripten versions can be found in the [Qt documentation](https://doc.qt.io/qt-6/wasm.html).
+
+Make sure you have a version of Qt for WebAssembly that will work, and clone the `emsdk` repository if you do not already have it.
+
+## Setting Up `emsdk`
+
+Once you know which Qt and Emscripten versions you will use, navigate to the root directory of the `emsdk` repo and run the following commands:
+
+```bash
+$ ./emsdk install <emscripten version>
+$ ./emsdk activate <emscripten version>
+$ source ./emsdk_env.sh
+```
+
+For example, if you are going to use Qt 6.4, the corresponding version of Emscripten is 3.1.14, so the first command will be:
+
+```bash
+$ ./emsdk install 3.1.14
+```
+
+On Windows, the third step, which sets up environment variables (`source` command above on Unix-like environments) is unnecessary because the required environment setup will already be done.
+
+## Building CXX-Qt
+
+When invoking CMake, the `CMAKE_TOOLCHAIN_FILE` variable needs to be set to the correct toolchain file; for example, if using Qt 6.4.2 on WebAssembly, the toolchain file is typically located at `/path/to/Qt/6.4.2/wasm_32/lib/cmake/Qt6/qt.toolchain.cmake`. This will set CMake up to use the correct Qt path, compiler, linker, and so forth.
+
+Generally, this does not need to be done manually. Using the `qt-cmake` binary bundled with your selected version of Qt WASM will set the toolchain file for you. However, in Qt 6.3 and below, the bundled CMake is version 3.22, while CXX-Qt requires at least version 3.24. For these versions of Qt, a more up-to-date CMake binary needs to be used to configure, so `CMAKE_TOOLCHAIN_FILE` needs to be passed into the `cmake` command.
+
+Navigate to the root directory of the CXX-Qt repo and run the following command to configure CXX-Qt for WebAssembly:
+
+```bash
+$ /path/to/qt-cmake -DBUILD_WASM=ON -B build .
+```
+
+If using a different CMake binary, instead do this:
+
+```bash
+$ <cmake binary> -DCMAKE_TOOLCHAIN_FILE=/path/to/qt.toolchain.cmake -DBUILD_WASM=ON -B build .
+```
+
+Finally, run `cmake --build` on the configured build directory to compile and link the project and examples. This can be any CMake binary, here the OS package works just fine:
+
+```bash
+$ cmake --build build
+```
+
+Then you can run the `qml_minimal` example like so:
+
+```bash
+$ emrun ./build/examples/qml_minimal/example_qml_minimal.html
+```
+
+## Working Examples
+
+Not all of the examples are currently supported for WASM builds.
+
+Example|Working
+-|-
+`qml-minimal-no-cmake`|❌ broken
+`demo_threading`|❌ broken
+`qml_features`|✅ working
+`qml_minimal`|✅ working
+
+For more information, see the [Known Issues](#known-issues) section at the bottom of this page.
+
+## Compiling CXX-Qt Projects for WebAssembly
```

Comment:
as discussed maybe this comes higher up and the bits about building the `cxx-qt` repository are later. Note that `-DCMAKE_TOOLCHAIN_FILE=/path/to/qt.toolchain.cmake` info should probably be added to this part so folks know how to build it in their project

Reply:
(or the `qt-cmake ` alternative route)

Reply:
Maybe that whole bit explaining the toolchain situation can be it's own section ?

So have these as separate sections?
- CMake toolchain fun
- Building your own project
- Building CXX-Qt repository / examples